jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -405,6 +435,29 @@ object XGBoost extends Serializable {
     xgBoostModel.setPredictionCol(predCol)
   }
 
+  private def loadPrevBooster(sparkContext: SparkContext, hdfsTmpPath: String):
```

Comment:
we may want to do some cleanup when loading the latest checkpoint, e.g. use an async thread to delete all old files. If failed, just put a warning